### PR TITLE
Spline comp fixes

### LINKED ISSUE #1207: Fixes for spline_comp
### Summary of Issue

Misc fixes to `spline_comp` and its documentation.

### Issue Type

- [x] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

- Make `We have included three different distribution functions for users to replicate functionality that used to be built-in to the individual akima and bsplines components` a note
- Add in the description of `num_cp` that if it’s None, it will use a linspace 
- Move spline_distribution into `om.api`
- Make sure to add the right imports for the doc cases 
- Add `units=y_units` and add test for it.
- Clean docs for spline_distribution 

### Example

N/A

### PR DESCRIPTION
**Pull request by DKilkenny**
### Summary

See issue 

### Related Issues

- Resolves #1207 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/api.py [hunk 1]
```diff
@@ -64,6 +64,9 @@
     MultiFiSurrogateModel
 
 from openmdao.utils.find_cite import print_citations
+from openmdao.utils.spline_distributions import cell_centered
```
Comment (Kenneth-T-Moore) [inline openmdao/api.py]: Check with jubs, but my interpretation of the Issue is that he wanted the module moved in, not the individual methods. So, you would do `om.spline_distributions.cell_centered` to get one.

File: openmdao/docs/features/building_blocks/components/spline_comp.rst [hunk 2]
```diff
@@ -132,24 +132,40 @@ specify the number of control points using the 'num_cp' argument.
...
-`sine_distribution` function creates a sinusoidal distribution, in which points are clustered towards the
-ends. A 'phase' argument is also included, and a phase of pi/2.0 clusters the points in the center with
-fewer points on the ends.
+The `cell_centered` function takes the number of cells, and the start and end values, and returns a
+vector of points that lie at the center of those cells. The 'node_centered' function reproduces the
+functionality of numpy's linspace.  Finally, the `sine_distribution` function creates a sinusoidal
+distribution, in which points are clustered towards the ends. A 'phase' argument is also included,
+and a phase of pi/2.0 clusters the points in the center with fewer points on the ends.
+
+.. note::
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/features/building_blocks/components/spline_comp.rst]: I would move the note up above the new paragraph (though again, verify with justin)

File: openmdao/components/interp_util/tests/test_interp_nd.py [hunk 3]
```diff
@@ -56,60 +57,24 @@ def test_interp_spline_akima_derivs(self):
```
Comment (Kenneth-T-Moore) [inline openmdao/components/interp_util/tests/test_interp_nd.py]: I am nervous about losing this original test of the derivative values.  I would copy this to a new test (put it in the Feature test object since it is in the docs) and then revert the changes here back to the original.

File: openmdao/docs/features/building_blocks/components/spline_comp.rst [hunk 4]
```diff
@@ -132,24 +132,40 @@ specify the number of control points using the 'num_cp' argument.
...
+    We have included three different distribution functions for users to replicate functionality that used to
+    be built-in to the individual akima and bsplines components.
 
 .. embed-code::
-    openmdao.components.tests.test_spline_comp.SplineCompFeatureTestCase.test_spline_distribution_example
+    openmdao.utils.spline_distributions.sine_distribution
```
Comment (swryan) [inline openmdao/docs/features/building_blocks/components/spline_comp.rst]: you're showing the implementations of the functions?
seems odd to put implementation details in the user docs...

Comment (DKilkenny) [reply openmdao/docs/features/building_blocks/components/spline_comp.rst]: Fixing

File: openmdao/api.py (hunk 1 above)
Comment (DKilkenny) [reply openmdao/api.py]: He's fine with this

File: openmdao/docs/features/building_blocks/components/spline_comp.rst (hunk 2 above)
Comment (DKilkenny) [reply openmdao/docs/features/building_blocks/components/spline_comp.rst]: He's fine with this

File: openmdao/components/interp_util/tests/test_interp_nd.py (hunk 3 above)
Comment (DKilkenny) [reply openmdao/components/interp_util/tests/test_interp_nd.py]: I'll add the original test to the non features test class. I don't think it's necessary for the user to see a huge output chunk
